ui/app/pages/confirm-decrypt-message/confirm-decrypt-message.component.js

```diff
@@ -0,0 +1,330 @@
+    })
+    clearConfirmTransaction()
+    cancelDecryptMessage(event)
+  }
+
+  _removeBeforeUnload = () => {
+    if (getEnvironmentType(window.location.href) === ENVIRONMENT_TYPE_NOTIFICATION) {
+      window.removeEventListener('beforeunload', this._beforeUnload)
+    }
+  }
+
+  copyMessage = () => {
+    copyToClipboard(this.state.rawMessage)
+    this.context.metricsEvent({
+      eventOpts: {
+        category: 'Transactions',
+        action: 'Decrypt Message Copy',
+        name: 'Copy',
+      },
+    })
+    this.setState({ hasCopied: true })
+    setTimeout(() => this.setState({ hasCopied: false }), 3000)
+  }
+
+  renderHeader = () => {
+    return (
+      <div className="request-decrypt-message__header">
+        <div className="request-decrypt-message__header-background" />
+
+        <div className="request-decrypt-message__header__text">
+          { this.context.t('decryptRequest') }
+        </div>
+
+        <div className="request-decrypt-message__header__tip-container">
+          <div className="request-decrypt-message__header__tip" />
+        </div>
+      </div>
+    )
+  }
+
+  renderAccount = () => {
+    const { selectedAccount } = this.state
+
+    return (
+      <div className="request-decrypt-message__account">
+        <div className="request-decrypt-message__account-text">
+          { `${this.context.t('account')}:` }
+        </div>
+
+        <div className="request-decrypt-message__account-item">
+          <AccountListItem
+            account={selectedAccount}
+            displayBalance={false}
+          />
+        </div>
+      </div>
+    )
+  }
+
+  renderBalance = () => {
+    const { balance, conversionRate } = this.props
+
+    const balanceInEther = conversionUtil(balance, {
+      fromNumericBase: 'hex',
+      toNumericBase: 'dec',
+      fromDenomination: 'WEI',
+      numberOfDecimals: 6,
+      conversionRate,
+    })
+
+    return (
+      <div className="request-decrypt-message__balance">
+        <div className="request-decrypt-message__balance-text">
+          { `${this.context.t('balance')}:` }
+        </div>
+        <div className="request-decrypt-message__balance-value">
+          { `${balanceInEther} ETH` }
+        </div>
+      </div>
+    )
+  }
+
+  renderRequestIcon = () => {
+    const { requesterAddress } = this.props
+
+    return (
+      <div className="request-decrypt-message__request-icon">
+        <Identicon
+          diameter={40}
+          address={requesterAddress}
+        />
+      </div>
+    )
+  }
+
+  renderAccountInfo = () => {
+    return (
+      <div className="request-decrypt-message__account-info">
+        { this.renderAccount() }
+        { this.renderRequestIcon() }
+        { this.renderBalance() }
+      </div>
+    )
+  }
+
+  renderBody = () => {
+    const { txData } = this.props
+
+    const origin = this.props.domainMetadata[txData.msgParams.origin]
+    const notice = this.context.t('decryptMessageNotice', ['[' + origin.name + ']'])
```

Comment:
```suggestion
    const notice = this.context.t('decryptMessageNotice', [origin.name)
```
These extra square brackets are not needed - the square brackets in the mockups/designs were just meant to indicate that the contents were variable.